Work orders manager enhancement (#3781)
DFHack/dfhack: docs/plugins/orders.rst

    manager orders. It will not clear the orders that already exist.
``orders clear``
    Deletes all manager orders in the current embark.

> ``orders reset``

Reviewer: I'd call this "recheck", not "reset". "reset" is too similar to "clear" in semantics.

Reviewer: and it might also be useful to differentiate between UI states -- `orders recheck` when viewing the details for a particular order could just reset that order, whereas it could reset all orders if called at any other time. alternately, the syntax could be `orders recheck [this]` where if `this` is specified, it will just reset the currently selected order, and all orders otherwise.

Author: Changed 'reset' to 'recheck'. 
We already have keybing for a single order on order conditions screen. Do you mean to rewrite it?

Reviewer: yes, I think it makes sense to merge that into the `orders` plugin.

Author: Added `this` keyword to recheck only selected order. Moved single order invalidation to the orders plugin.

    Invalidates manager orders forcing material conditions recheck.
``orders sort``
    Sorts current manager orders by repeat frequency so repeating orders don't
    prevent one-time orders from ever being completed. The sorting order is:
    one-time orders first, then yearly, seasonally, monthly, and finally, daily.
``orders sort_type``
    Sorts current manager orders by job type, making it easier to locate orders
    that produce similar items. The sorting is done by reaction name, job type
    and item subtype. If orders are equal by these fields the sorting falls back
    to sort by frequency.
``orders sort_material``
    Sorts current manager orders by material, making it easier to locate orders
    that produce items of the same material. The sorting is done by material type
    and material category. If orders are equal by these fields the sorting falls back

>     to sort by job type.

Reviewer: I am very wary of using these to sort the base orders list. The reason for the existing sort is to work around deficiencies in the vanilla order processing code. any other sort order risks making some orders non-completable.

What I was thinking about instead was an alternate UI where you could sort, filter, and group orders by any criteria. the organization would be saved separately from the vanilla orders list, which would stay in the existing sorted order for safety.

Author: Well, they are not sorting automatically. This is not a problem at all if orders-sort enabled. 
About UI, I think that this is a great idea. I just don't think I can write something big right now.

Reviewer: I think remove them until we can add a proper UI layer. Could you add the code as a comment at https://github.com/DFHack/dfhack/issues/3555 so your work isn't lost?

Author: Removed these functions and posted them in a comment.

    manager orders. It will not clear the orders that already exist.
``orders clear``
    Deletes all manager orders in the current embark.
``orders recheck [this]``

>     Sets the status to ``Checking`` (from ``Active``) of all work orders or one
>     currently viewed if 'this' option is passed. Work order conditions screen 
>     should be open in this case. This makes the manager reevaluate its conditions.

Reviewer: ```suggestion
    Sets the status to ``Checking`` (from ``Active``) for all work orders. if the
    "this" option is passed, only sets the status for the workorder whose condition
    details page is open. This makes the manager reevaluate its conditions.
```
